src/prefect/deployments.py

```diff
@@ -68,11 +70,47 @@ async def run_deployment(
     if scheduled_time is None:
         scheduled_time = pendulum.now("UTC")
 
+    parameters = parameters or {}
+
     deployment = await client.read_deployment_by_name(name)
+
+    flow_run_ctx = FlowRunContext.get()
+    if flow_run_ctx:
+        # This was called from a flow. Link the flow run as a subflow.
+        from prefect.engine import (
+            Pending,
+            _dynamic_key_for_task_run,
+            collect_task_run_inputs,
+        )
+
+        task_inputs = {
+            k: await collect_task_run_inputs(v) for k, v in parameters.items()
+        }
+
+        # Generate a task in the parent flow run to represent the result of the subflow
+        dummy_task = Task(
```

Comment:
not that it matters but I always thought that "virtual tasks" are a better name for these

Reply:
:) something to consider when we create an abstract function for this logic